README.md

```diff
@@ -1,13 +1,213 @@
 [![Build Status](https://travis-ci.org/scalameta/lsp4s.svg?branch=master)](https://travis-ci.org/scalameta/lsp4s)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12/badge.svg)](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12)
 
 # lsp4s
-Scala implementation for JSON-RPC and Language Server Protocol
+
+This is a Scala implementation for [JSON-RPC][] and the [Language Server
+Protocol][lsp] with a focus on compile-time safety and extensibility. This
+project contains two modules:
+
+- `jsonrpc`: core data structures and client/server implementations for JSON-RPC
+  with additional support for cancellation via method `$/cancelRequest`.
+- `lsp4s`: data structures for the Language Server Protocol.
+
+The `jsonrpc` module has the following dependencies:
+
+- [Monix][]: for asynchronous programming primitives `Task` and `Observable`.
+  Monix `Task` was chosen over the standard library `Future` in order to support
+  cancellation, which is an important part of the Language Server Protocol.
+  Monix `Observable` is also a good fit the stream-based nature of the JSON-RPC
+  protocol.
+- [Circe][]: for JSON serialization and de-serialization. Our requirements for
+  JSON include a parser, printer, custom decoding/encoding APIs as well as
+  automatic (or semi-automatic) derivation for case classes. Circe was chosen
+  over alternative JSON libraries for its high quality error handling and
+  type-safe APIs with strong compile-time guarantees.
+- [Scribe][]: for logging. Scribe was chosen over alternative logging frameworks
+  (such as slf4j) for its usage of compile-time reflection over runtime
+  reflection in addition to a fully programmable configuration API avoiding the
+  need for XML.
+
+The `lsp4s` module has the following dependencies:
+
+- `jsonrpc`: the module in this project
+- [Enumeratum]: for reflection-free and boilerplate-free enumerations.
+
+[enumeratum]: https://github.com/lloydmeta/enumeratum
+[json-rpc]: http://www.jsonrpc.org
+[lsp]: https://microsoft.github.io/language-server-protocol/
+[monix]: https://monix.io/
+[circe]: https://circe.github.io/circe/
+[scribe]: https://github.com/outr/scribe
+
+## Getting Started
+
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12/badge.svg)](https://maven-badges.herokuapp.com/maven-central/org.scalameta/lsp4s_2.12)
+
+To implement a language server, add the following dependency you need.
```

Comment:
This sounds like Yoda 😅

Reply:
Haha, good catch, was a bad rephrase from an old text